src/pan3d/explorers/analytics.py

```diff
@@ -210,48 +206,42 @@ def _build_ui(self, **kwargs):
                         source=self.source, toggle="chart_expanded"
                     )
 
+        self.ctrl.source_update_rendering_panel = self.rendering.update_from_source
```

Comment:
Attaching method controller is usually better to keep them close to the widget definition, so when refactoring you have no risk of doing half of the work.